corehq/apps/accounting/static/accounting/js/confirm_plan.js

```diff
@@ -17,9 +17,51 @@ hqDefine('accounting/js/confirm_plan', [
         self.currentPlan = currentPlan;
         self.newPlan = newPlan;
 
-        // If the user is upgrading, don't let them continue until they have agreed to the therms
+        // If the user is upgrading, don't let them continue until they agree to the minimum subscription terms
         self.userAgreementSigned = ko.observable(!isUpgrade);
 
+        self.downgradeReasonList = ko.observableArray([
+            "My project ended",
+            "The funding for my project ended",
+            "I don’t need the features of my paid plan anymore but I plan on continuing using CommCare",
+            "We are switching to a different mobile data collection tool",
+        ]);
+        self.newToolReasonList = ko.observableArray([
+            "For budget reason",
+            "I need more limited features",
+            "I need additional/custom features",
+            "Other",
+        ]);
+
+        self.downgradeReason = ko.observable("");
+        self.newTool = ko.observable("");
+        self.newToolReason = ko.observable("");
+        self.otherNewToolReason = ko.observable("");
+        self.requiredQuestionsAnswered = ko.observable(false);
```

Comment:
fyi there's a knockout validation plugin in our codebase - you'd add `@use_ko_validation` to the python view, use `.extend` on the required observables as described in [the docs](https://github.com/Knockout-Contrib/Knockout-Validation#getting-started), and use our custom [koValidationStateFeedback`(https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/validators.ko.js) binding to display messages. Totally optional; I haven't used it enough to be sure whether it would be overkill for the amount of validation here or would be an improvement.

Reply:
This does seem really cool, and I'm playing with it now! But, I don't think that it's necessary for this change right now.